### PR TITLE
Update ByteMLPerf with vendor Moffett-AI (S30) and backend SPU.

### PR DESCRIPTION
**Pull request by moffett-ai**

### CODE REVIEW

File: byte_mlperf/backends/SPU/compile_backend_spu.py [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+# limitations under the License.
+
+
+import os
+import json
+import logging
+import time
+from typing import Any, Dict, List, Optional
+from byte_mlperf.backends import compile_backend
+from base_compile import Resnet50Builder, BertBaseBuilder, AlbertBuilder, RobertaBuilder, ConformerBuilder
```
Comment (Minghui-BD) [inline byte_mlperf/backends/SPU/compile_backend_spu.py:22]: base_compiler是闭源的部分么？

File: byte_mlperf/backends/SPU/compile_backend_spu.py [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+        dump_dir = self.interact_info["dump_dir"]
+        dataset_dir = self.interact_info["dataset_dir"]
+        dataset_cfg = self.interact_info["dataset_cfg"]
+        dtype = self.interact_info["dtype"]
+        batch_size = self.interact_info["batch_size"]
+        verify = self.interact_info["verify"]
+
+        start_time = time.time()
+        builder = sparser_builder(onnx_path, dump_dir, dataset_dir, dataset_cfg, dtype, batch_size, verify)
+        result = builder()
```
Comment (Minghui-BD) [inline byte_mlperf/backends/SPU/compile_backend_spu.py:71]: result里补充下稀疏倍率吧

File: byte_mlperf/reports/SPU/bert-torch-fp32/bert-torch-fp32.png
Comment (Minghui-BD) [inline byte_mlperf/reports/SPU/bert-torch-fp32/bert-torch-fp32.png:1]: 这个backend的分布看着很奇怪，check下？

File: byte_mlperf/reports/SPU/roberta-torch-fp32/roberta-torch-fp32.png
Comment (Minghui-BD) [inline byte_mlperf/reports/SPU/roberta-torch-fp32/roberta-torch-fp32.png:1]: 这个也是，是不是过拟合了。。。？

File: byte_mlperf/backends/SPU/runtime_backend_spu.py [hunk 3]
```diff
@@ -0,0 +1,263 @@
+import logging
+import tensorflow as tf
+import torch
+import numpy as np
+import time
+import yaml
+import multiprocessing
+from byte_mlperf.backends import runtime_backend
+import spu_backend
```
Comment (Minghui-BD) [inline byte_mlperf/backends/SPU/runtime_backend_spu.py:9]: spu_backend & infrence 这倆文件也是闭源的么?

File: byte_mlperf/backends/SPU/compile_backend_spu.py [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+        end_time = time.time()
+        compilation_time = round(end_time - start_time, 2)
+        assert isinstance(result, dict), "return is not dict"
+        result.update({"compilation_time": compilation_time})
+        return result
+
+    def get_interact_profile(self, configs: Dict[str, Any]):
+        model_profile = []
+        file_path = f"backends/{configs['model']}.json"
```
Comment (JianzheXiao) [inline byte_mlperf/backends/SPU/compile_backend_spu.py:80]: 这个路径下不会有json file存在吧

File: byte_mlperf/backends/SPU/compile_backend_spu.py [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+            "conformer-encoder-onnx-fp32": ConformerBuilder
+        }
+
+        if name in builder_dict:
+            sparser_builder = builder_dict[name]
+        else:
+            raise NotImplementedError(f"task: {name} not supported")
+
+        self.interact_info = self.get_interact_profile(configs)
+        onnx_path = self.interact_info["onnx_path"]
```
Comment (JianzheXiao) [inline byte_mlperf/backends/SPU/compile_backend_spu.py:61]: 目前这些config我好像没在spu.json里面看到？

File: byte_mlperf/backends/SPU/runtime_backend_spu.py [hunk 6]
```diff
@@ -0,0 +1,263 @@
...
+    def benchmark(self, dataloader):
+        batch_sizes = self.workload['batch_sizes']
+        reports = []
+        if self.model and self.model_name != "resnet50-torch-fp32":
+            self.model.initialize()
+        for idx, batch_size in enumerate(batch_sizes):
+            if batch_size != self.batch_size:
+                continue
+            iterations = self.workload['iterations'][idx]
+            buffer_depth = self.workload['buffer_depth'][idx]
```
Comment (JianzheXiao) [inline byte_mlperf/backends/SPU/runtime_backend_spu.py:90]: 这些参数是不是会对workloads作修改？

File: byte_mlperf/reports/SPU/bert-torch-fp32/bert-torch-fp32.png
Comment (moffett-ai) [reply byte_mlperf/reports/SPU/bert-torch-fp32/bert-torch-fp32.png:1]: squad_postprocess 用了量化的 sigmoid 结果，跟 CPU 的 logit 直接比较不合适，这里调整下再MR一版。

File: byte_mlperf/reports/SPU/roberta-torch-fp32/roberta-torch-fp32.png
Comment (moffett-ai) [reply byte_mlperf/reports/SPU/roberta-torch-fp32/roberta-torch-fp32.png:1]: squad_postprocess 用了量化的 sigmoid 结果，跟 CPU 的 logit 直接比较不合适，这里调整下再MR一版。

File: byte_mlperf/backends/SPU/compile_backend_spu.py (hunk 1 above)
Comment (moffett-ai) [reply byte_mlperf/backends/SPU/compile_backend_spu.py:22]: 是。

File: byte_mlperf/backends/SPU/runtime_backend_spu.py (hunk 3 above)
Comment (moffett-ai) [reply byte_mlperf/backends/SPU/runtime_backend_spu.py:9]: 是的。

File: byte_mlperf/backends/SPU/compile_backend_spu.py (hunk 2 above)
Comment (moffett-ai) [reply byte_mlperf/backends/SPU/compile_backend_spu.py:71]: OK. Will add this info in result.json.

File: byte_mlperf/backends/SPU/compile_backend_spu.py (hunk 4 above)
Comment (moffett-ai) [reply byte_mlperf/backends/SPU/compile_backend_spu.py:80]: 我们会把放置 model.json 的地方调整到 backends/SPU/interact_info/*.json 提交一版。

File: byte_mlperf/backends/SPU/compile_backend_spu.py (hunk 5 above)
Comment (moffett-ai) [reply byte_mlperf/backends/SPU/compile_backend_spu.py:61]: 没有用到spu.json,  我们会把编译配置放到 backends/SPU/interact_info/[model_name].json 中。

File: byte_mlperf/backends/SPU/runtime_backend_spu.py (hunk 6 above)
Comment (moffett-ai) [reply byte_mlperf/backends/SPU/runtime_backend_spu.py:90]: 实际上不需要动到 workload 里的东西，这部份我们改一下提交。

File: byte_mlperf/backends/SPU/runtime_backend_spu.py (hunk 3 above)
Comment (Minghui-BD) [reply byte_mlperf/backends/SPU/runtime_backend_spu.py:9]: > 是的。

请教下，这部分闭源是啥考虑？如果有闭源的必要话，麻烦在README补充下如何获得这些文件
